### ISSUE TITLE
Only CPUExecutionProvider is available in ONNX Runtime InferenceSession

### ISSUE THREAD

**Issue by grazder** [microsoft/onnxruntime#9451]
**Problem**
We are trying to infer models with onnxruntime 1.9.0 and CUDA 11.1. But whenever we are trying to get providers of our models we get only `['CPUExecutionProvider']`.

It worked well with CUDA 10.2 and onnxruntime 1.6.0, but now we get `['CPUExecutionProvider']` instead of `['CUDAExecutionProvider', 'CPUExecutionProvider']` we had before.

**To Reproduce**
We've created a simple Dockerfile to reproduce this bug.

```
FROM nvidia/cuda:11.1.1-devel-ubuntu20.04

ENV LANG=C.UTF-8 LC_ALL=C.UTF-8
ENV DEBIAN_FRONTEND=noninteractive
ENV PATH /opt/conda/bin:/usr/local/cuda-11.1/bin:$PATH

# Package update/instalation
RUN apt update --fix-missing -y && apt upgrade -y
RUN apt install -y \
        sudo \
        wget

# Miniconda installation
RUN wget --quiet https://repo.anaconda.com/miniconda/Miniconda3-py39_4.9.2-Linux-x86_64.sh -O ~/miniconda.sh && \
    /bin/bash ~/miniconda.sh -b -p /opt/conda && \
    rm ~/miniconda.sh

# PyTorch installation
RUN conda install -c pytorch -c nvidia pytorch cudatoolkit=11.1

RUN pip install -U onnx onnxruntime-gpu

# Conda cleaning
RUN conda clean -afy
```
Inside this container we ran folowing code:
```
python -c 'import torch; print(torch.version.cuda, torch.backends.cudnn.version())'
```
Out: `11.1 8005`
```
python -c 'import onnxruntime; print(onnxruntime.get_device())'
```
Out: `GPU`
```
python -c "import onnxruntime; session = onnxruntime.InferenceSession('fp16_model.onnx'); print(session.get_providers())"
```
Out: `['CPUExecutionProvider']`

But on CUDA - 10.2 & onnxruntime - 1.6.0 setup we had `['CUDAExecutionProvider', 'CPUExecutionProvider']` in the last line output.

Link to a simple model ONNX we tested: https://disk.yandex.com/d/nTCPSCn67eL9sQ

**Workaround**
To make it work we rebuilt the ORT by ourselves using CUDNN - 11.4

**Comment by skottmckay** [comment]
Was cudnn installed in your docker setup? I believe that's a required dependency but not sure if cudatoolkit includes that. 

Note that pytorch may statically link to cudnn so checking torch.backends.cudnn.version doesn't necessarily mean that ORT has the files it requires.

**Comment by grazder** [comment]
Thank you for your comment! We checked that this base image `nvidia/cuda:11.1.1-devel-ubuntu20.04` does not install cudnn by default. As a solution, we found the image `nvidia/cuda:11.1.1-cudnn8-runtime-ubuntu20.04` with the same cuda and ubuntu version but cudnn8 installed.

We ran
```
python -c "import onnxruntime; session = onnxruntime.InferenceSession('fp16_model.onnx'); print(session.get_providers())"
```
in this setup and got
Out: `['CUDAExecutionProvider', 'CPUExecutionProvider']`
